app/Http/Requests/PostRequest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Rogue\Http\Requests;
+
+class PostRequest extends Request
+{
+    protected $rules = [
+        'northstar_id' => 'required|string',
+        'campaign_id' => 'required|int',
+        'campaign_run_id' => 'required|int',
+    ];
+
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        if (! isset($this->request->event_type)) {
+            return [
+                'event_type' => 'required|in:post_photo',
+            ];
+        }
+
+        return $this->getRules($this->request);
+    }
+
+    /**
+     * Determines the validation ruleset to use for a particular type of post.
+     *
+     * @param  object $request
+     * @return array|null
+     */
+    protected function getRules($request)
+    {
+        switch ($request->event_type) {
+            case 'post_photo':
+                return array_merge($this->rules, $this->setPhotoRules());
```

Comment:
I'm curious why `$this->rules` is a property but `$this->setPhotoRules()` is a function. If you'd like to keep it as a function, would it make sense to rename it to `$this->getPhotoRules()` since it doesn't mutate anything?